src/sentry/api/bases/organization.py

```diff
@@ -91,6 +103,116 @@ class OrganizationDiscoverSavedQueryPermission(OrganizationPermission):
 
 class OrganizationEndpoint(Endpoint):
     permission_classes = (OrganizationPermission, )
+    # Defines what happens if no date filter parameters are passed. If False,
+    # no date filtering occurs. If True, we provide default values.
+    DATE_FILTER_OPTIONAL = False
+
+    def get_project_ids(self, request, organization, force_global_perms=False):
+        """
+        Determines which project ids to filter the endpoint by. If a list of
+        project ids is passed in via the `project` querystring argument then
+        validate that these projects can be accessed. If not passed, then
+        return all project ids that the user can access within this
+        organization.
+
+        :param request:
+        :param organization: Organization to fetch projects for
+        :param force_global_perms: Permission override. Allows subclasses to
+        perform their own validation and allow the user to access any project
+        in the organization.
+        :return: A list of project ids, or raises PermissionDenied.
+        """
+        project_ids = set(map(int, request.GET.getlist('project')))
+
+        requested_projects = project_ids.copy()
+
+        om_role = None
+        if request.user.is_authenticated():
+            try:
+                om_role = OrganizationMember.objects.filter(
+                    user=request.user,
+                    organization=organization,
+                ).values_list('role', flat=True).get()
+            except OrganizationMember.DoesNotExist:
+                pass
+
+        if (
+            request.user.is_superuser
+            or (om_role and roles.get(om_role).is_global)
+            or organization.flags.allow_joinleave
+            or force_global_perms
```

Comment:
I added some extra permission things here based on what I saw on the release endpoint